internal/common/types.go

```diff
@@ -90,6 +90,16 @@ type DeviceInfo struct {
 	// UpdateLastConnected specifies whether to update device's LastConnected
 	// timestamp in metadata.
 	UpdateLastConnected bool
+
+	Discovery DiscoveryInfo
+}
+
+// DiscoveryInfo is a struct which contains configuration of device auto discovery.
+type DiscoveryInfo struct {
+	// Enabled controls whether or not device discovery is enabled.
+	Enabled bool
+	// Interval indicates how often the discovery process will be triggered in seconds.
+	Interval int64
```

Comment:
I know you mentioned that this is an int64 because that's what a Duration is. Is there a reason you can't just make this a Duration then, or does that causes problems with loading the configuration?